### PR TITLE
fix: Check if the requirements config was read from the existing file

### PR DESCRIPTION
**Pull request by hferentschik**
based on PR #6658

### CODE REVIEW

**Comment by hferentschik** [comment]
@RafalKorepta, down to one BDD test failure. How this change relates to the Spring demo app not coming up, I'm currently not sure yet.

@abayer, any idea what the problem could be. The failure is in this case not really so helpful.

**Comment by abayer** [comment]
I’d guess that the staging env pipeline is failing, either in PR or release. Only way to figure that out is probably to jump on the failed cluster before it gets deleted and check the logs.

File: pkg/config/install_requirements_test.go [hunk 1]
```diff
@@ -356,3 +357,53 @@ func TestHelmfileRequirementValues(t *testing.T) {
...
+		{"a/b/c", false, false},
+		{"e", true, true},
+		{"e/f", true, true},
+		{"e/f/g", true, false},
+	}
+
+	for _, testCase := range gitPublicTests {
+		t.Run(testCase.requirementsPath, func(t *testing.T) {
+			dir, err := ioutil.TempDir("", "jx-test-load-requirements-config")
+			assert.NoError(t, err, "failed to create tmp directory")
```
Comment (RafalKorepta) [inline pkg/config/install_requirements_test.go]: NIT: `require.NoError` can stop the unit test. In my experience set up should fail fast and assertion should be used after calling the function under test.

File: pkg/config/install_requirements_test.go [hunk 2]
```diff
@@ -356,3 +357,53 @@ func TestHelmfileRequirementValues(t *testing.T) {
...
+			defer func() {
```
Comment (RafalKorepta) [inline pkg/config/install_requirements_test.go]: NIT: I sometimes use the following snippet:
```
defer assert.NoError(t, os.RemoveAll(dir))
```

File: pkg/config/install_requirements_test.go [hunk 3]
```diff
@@ -356,3 +357,53 @@ func TestHelmfileRequirementValues(t *testing.T) {
...
+			testPath := filepath.Join(dir, testCase.requirementsPath)
+			err = os.MkdirAll(testPath, 0700)
+			assert.NoError(t, err, "unable to create test path %s", testPath)
+
+			var expectedRequirementsFile string
+			if testCase.createRequirements {
+				expectedRequirementsFile = filepath.Join(testPath, config.RequirementsConfigFileName)
+				dummyRequirementsData := []byte("webhook: prow\n")
+				err := ioutil.WriteFile(expectedRequirementsFile, dummyRequirementsData, 0644)
+				assert.NoError(t, err, "unable to write requirements file %s", expectedRequirementsFile)
```
Comment (RafalKorepta) [inline pkg/config/install_requirements_test.go]: NIT: `require.NoError`

File: pkg/config/install_requirements_test.go [hunk 4]
```diff
@@ -356,3 +357,53 @@ func TestHelmfileRequirementValues(t *testing.T) {
...
+		t.Run(testCase.requirementsPath, func(t *testing.T) {
+			dir, err := ioutil.TempDir("", "jx-test-load-requirements-config")
+			assert.NoError(t, err, "failed to create tmp directory")
+			defer func() {
+				_ = os.RemoveAll(dir)
+			}()
+
+			testPath := filepath.Join(dir, testCase.requirementsPath)
+			err = os.MkdirAll(testPath, 0700)
+			assert.NoError(t, err, "unable to create test path %s", testPath)
```
Comment (RafalKorepta) [inline pkg/config/install_requirements_test.go]: NIT: `require.NoError`

File: pkg/config/install_requirements_test.go [hunk 5]
```diff
@@ -356,3 +357,53 @@ func TestHelmfileRequirementValues(t *testing.T) {
...
 	assert.FileExists(t, valuesFile, "file %s should exist", valuesFile)
 }
+
+func Test_LoadRequirementsConfig(t *testing.T) {
+	t.Parallel()
+
+	var gitPublicTests = []struct {
+		requirementsPath   string
+		createRequirements bool
+		loadSuccessful     bool
```
Comment (RafalKorepta) [inline pkg/config/install_requirements_test.go]: This is an unused filed. Or am I missing something?

Comment (hferentschik) [reply pkg/config/install_requirements_test.go]: Nope, not at all. I had initially something else in mind, but in the end, never needed this flag. Going to remove it.

File: pkg/config/install_requirements_test.go (hunk 4 above)
Comment (hferentschik) [reply pkg/config/install_requirements_test.go]: Interesting, what's the difference? Is there a semantic difference? I think _require_ crossed my path at some stage, but I usually use assert. Well, personally I prefer to write a Gingko test if start a new test ;-)

File: pkg/config/install_requirements_test.go (hunk 1 above)
Comment (hferentschik) [reply pkg/config/install_requirements_test.go]: ahhh, sorry missed this. that's the explanation first time around

Comment (hferentschik) [reply pkg/config/install_requirements_test.go]: Good one. I agree with your reasoning. I prefer to fail fast as well. I'll remember this for the future.

**Comment by hferentschik** [comment]
@RafalKorepta, thanks for the review. Much appreciated. I cleaned this up a bit. Let's hope it passes  all tests.

**Pull request merged**